### ISSUE TITLE
peacocktv.com: detection

### ISSUE THREAD

**Issue by comfortablemilk4454** [uBlockOrigin/uAssets#21255]
### Prerequisites

- [X] This is **NOT** a [YouTube](https://www.github.com/uBlockOrigin/uAssets/issues/20586), [Facebook](https://www.github.com/uBlockOrigin/uAssets/issues/3367) or [Twitch](https://www.github.com/uBlockOrigin/uAssets/issues/5184) report.
- [X] I read and understand the [policy about what is a valid filter issue](https://github.com/uBlockOrigin/uAssets/blob/master/README.md#uassets).
- [X] I verified that this issue is not a duplicate. (Search [here](https://github.com/uBlockOrigin/uAssets/issues) to find out.)
- [X] I forced an update of my filter lists. (Click the ["Purge all caches"](https://github.com/gorhill/uBlock/wiki/Dashboard:-Filter-lists#purge-all-caches) button while holding the 'Shift' key, then click the ["Update now"](https://github.com/gorhill/uBlock/wiki/Dashboard:-Filter-lists#update-now) button.)
- [X] I did not remove any of the [default filter lists](https://github.com/uBlockOrigin/uAssets/assets/66517106/6e8d024a-5869-47cf-b142-b2357eb6c7d0), or I have verified that the issue was not caused by removing any of the default lists.
- [X] I did not enable additional filter lists, or I have verified that the issue still occurs without enabling additional filter lists.
- [X] I do not have custom filters/rules, or I have verified that the issue still occurs without custom filters/rules.
- [X] I am not using uBlock Origin (uBO) along with other content blocker extensions.
- [X] I have verified that the web browser's built-in blocker or [DNS blocking](https://en.wikipedia.org/wiki/DNS_blocking) (standalone or through a VPN) is not causing the issue.
- [ ] I did not answer truthfully to **ALL** the above checkpoints.

### URL(s) where the issue occurs.

```
https://www.peacocktv.com/watch/home?orig_ref=direct
https://www.peacocktv.com/watch/playback/vod/GMO_00000000007105_01/056dba41-271c-36d6-a3be-82ca595197cf
```

### Description

so i have the no ads peacock plan but peacock is still asking me to disable uBO. see images for my filter lists, and see images for the popup on peacock. i made a [post on reddit](https://www.reddit.com/r/uBlockOrigin/comments/18773sh/ubo_peacock/) about this but since it hasnt been responded to for some time i thought id file an issue report here. thanks!

### Other extensions used

https://github.com/uBlockOrigin/uAssets/assets/43555540/616a82a5-94e2-4717-ab4f-849f7264505d

& windscribe vpn (But the issue still happens when i disable the vpn)

### Screenshot(s)

<details>
<summary>Screenshot(s)</summary>

![image](https://github.com/uBlockOrigin/uAssets/assets/43555540/c15391d0-39fd-4191-9ee8-de00227de825)
![image](https://github.com/uBlockOrigin/uAssets/assets/43555540/251aa5c9-d512-4420-aed0-eeab8756935b)
![image](https://github.com/uBlockOrigin/uAssets/assets/43555540/9a34fb4d-ec72-4293-96c3-d8a1bec6ac39)

</details>

### Configuration

<details>
<summary>Details</summary>

```yaml
uBlock Origin: 1.54.0
Firefox: 120
filterset (summary):
 network: 133893
 cosmetic: 233898
 scriptlet: 43582
 html: 1127
listset (total-discarded, last-updated):
 added:
  adguard-generic: 80635-4639, 9m
  adguard-social: 22079-869, 9m
  adguard-cookies: 28267-62, 9m
  ublock-cookies-adguard: 637-0, 9m
  adguard-popup-overlays: 26552-1561, 9m
  adguard-mobile-app-banners: 4825-1069, 9m
  adguard-other-annoyances: 14032-270, 9m
  adguard-widgets: 2355-1, 9m
  easylist-annoyances: 4234-121, 9m
  easylist-chat: 168-12, 9m
  fanboy-cookiemonster: 49131-3833, 9m
  ublock-cookies-easylist: 637-637, 9m
  [3 lists not shown]: [too many]
 default:
  user-filters: 1-0, never
  ublock-filters: 36695-339, 8m
  ublock-badware: 7557-11, 8m
  ublock-privacy: 924-4, 8m
  ublock-unbreak: 2202-0, 8m
  ublock-quick-fixes: 173-2, 8m
  easylist: 75886-2402, 9m
  easyprivacy: 33073-887, 9m
  urlhaus-1: 10190-0, 9m
  plowe-0: 3776-1, 9m
filterset (user): [array of 1 redacted]
userSettings:
 advancedUserEnabled: true
hiddenSettings:
 userResourcesLocation: [redacted]
supportStats:
 allReadyAfter: 294 ms (selfie)
 maxAssetCacheWait: 144 ms
popupPanel:
 blocked: 35
 network:
  conviva.com: 3
  cookielaw.org: 3
  demdex.net: 6
  freewheel.tv: 1
  nbcuni.com: 1
  newrelic.com: 2
  omtrdc.net: 16
  scorecardresearch.com: 3
 extended:
  ##.adsbox:remove()

```
</details>

**Comment by mapx-** [comment]
- test using only uBO, no other extension enabled, the default lists
- if still getting detection will need an account to further investigate the issue => if you agree to give us your account credentials I'll provide an email address where to send them

**Comment by mapx-** [comment]
email address: `axis.trivia@gmail.com`

**Comment by AIDaN80112** [comment]
just before i do that how do i know this is a legit email address for the uBO team?

**Comment by AIDaN80112** [comment]
oh sorry i dont know how to use github lol what shows that you're a member?

**Comment by mapx-** [comment]
![image](https://github.com/uBlockOrigin/uAssets/assets/10303732/6c39b529-75b1-4592-8e34-013cd8689488)

**Comment by AIDaN80112** [comment]
<img width="1022" alt="image" src="https://github.com/uBlockOrigin/uAssets/assets/43555540/3d62fd5a-ff0e-4427-aef1-38d2f545a8ac">
lol

**Comment by AIDaN80112** [comment]
oh i see it says you edited my op that must mean youre a part of the team sorry i just wanted to make sure u were legit before i send u my password lol. sending the email rn

**Comment by AIDaN80112** [comment]
i apologize my email isnt working rn apple mail for macOS sonoma 14.2 rc2 wont let me log in :(

**Comment by AIDaN80112** [comment]
restarting fixed it sending rn

**Comment by AIDaN80112** [comment]
just went through on my end

**Comment by AIDaN80112** [comment]
just sent it again, did it work this time?

**Comment by mapx-** [comment]
yeah

**Comment by AIDaN80112** [comment]
awesome

**Comment by mapx-** [comment]
add in "my filters" and test on your side:

`peacocktv.com##+js(rpnt, script, "enableAdBlockerDetection":true, "enableAdBlockerDetection":false)`

**Comment by AIDaN80112** [comment]
<img width="1624" alt="image" src="https://github.com/uBlockOrigin/uAssets/assets/43555540/38fd37d0-b2d3-41c0-bfe4-ba131257eb02">
<img width="829" alt="image" src="https://github.com/uBlockOrigin/uAssets/assets/43555540/f49ebe72-9663-446b-baeb-badb074ef269">
<img width="1624" alt="image" src="https://github.com/uBlockOrigin/uAssets/assets/43555540/a6a6c81a-bb32-40b7-a125-3ac8113e4dcc">

**Comment by mapx-** [comment]
I'll test in firefox too, seems working in chrome

**Comment by AIDaN80112** [comment]
i see.

**Comment by AIDaN80112** [comment]
yes it is only an annoyance peacock still lets you watch the content with uBO enabled

**Comment by mapx-** [comment]
remove the filter above, add & test:

`*$image,domain=peacocktv.com,redirect-rule=1x1.gif`

**Comment by AIDaN80112** [comment]
do i keep the ublock- user- value ![image](https://github.com/uBlockOrigin/uAssets/assets/43555540/339e845a-c05c-43c0-a24b-4e1d82cd38a3) in addition to removing the 2nd filter 
![image](https://github.com/uBlockOrigin/uAssets/assets/43555540/f3f39a6d-e9c0-4f30-824a-3aa33688a0e9)
 & adding the *$image,domain=peacocktv.com,redirect-rule=1x1.gif ?

**Comment by AIDaN80112** [comment]
just tested the way you described & still showing the adblock detector popup 
<img width="1624" alt="image" src="https://github.com/uBlockOrigin/uAssets/assets/43555540/c31ead33-091d-470a-b012-8a95da2a09bf">

**Comment by mapx-** [comment]
- working fine on my side
- disable any other addon / extension, disable in firefox `enhanced tracking protection`.
- you could try in some chromium browser too

**Comment by mapx-** [comment]
- go to about:preferences#privacy
- choose custom
- disable all the entries

**Comment by AIDaN80112** [comment]
oh ok, just did that & its still showing the disable ur adblocker thing

**Comment by mapx-** [comment]
@MasterKia see on slack the credentials if you can test using a US IP

**Comment by mapx-** [comment]
I added another filter, @AIDaN80112 be sure you have enabled uBO annoyances (update that list in 5-10 minutes and test again)

**Comment by AIDaN80112** [comment]
thank you! oh and can i re enable enhanced tracking protection now? also is this one the uBO annoyances?
<img width="554" alt="image" src="https://github.com/uBlockOrigin/uAssets/assets/43555540/ad9559a4-ac1f-4473-a075-c43f21b7737a">

**Comment by mapx-** [comment]
it's something on your side, follow all these steps:

- clear cookies / localStorage/ cache for the tested page
- reset uBO, don't modify any uBO setting, update all the filter lists, don't add any other list (other than your regional list)
- disable any custom filter for that site
- disable any other extension / addon in your browser, disable builtin blockers /shields/ tracking protection in brave/ opera/ vivaldi/ edge/ firefox etc
- if Opera browser: open opera://extensions/ and check the "allow access to search page results" option
- disable (if any) the web shields in your antivirus / antimalware software / other blocking software
- are you using a DNS which features adblocking (like AdGuard DNS)?=> test using another DNS (google for example =>  IP addresses 8.8.8.8 & 8.8.4.4)
- restart the system, reopen the browser, test again.

**Comment by mapx-** [comment]
it's why I asked you to test in some chromium browser, FF is pretty slow.

**Comment by AIDaN80112** [comment]
wym firefox is slow? oh sorry i thought firefox was chromium i didnt understand

**Comment by AIDaN80112** [comment]
its still not working and i just made a fresh browser profile on firefox. could it be because im on a mac?

**Comment by AIDaN80112** [comment]
@mapx-

**Comment by mapx-** [comment]
I have no idea about mac and I cannot test in firefox on my vpn (getting that error)

**Comment by AIDaN80112** [comment]
i see.

**Comment by AIDaN80112** [comment]
yeah its defo an issue with uBO & mac cuz i just tested it on chrome too with a fresh profile & fresh installation of uBO with default settings + enabling the "uBlock filters – Annoyances" filter list and its still happening in chrome

<img width="1624" alt="image" src="https://github.com/uBlockOrigin/uAssets/assets/43555540/0ef870c5-147f-4239-b693-d28c273495e3">

**Comment by AIDaN80112** [comment]
> I have no idea about mac and I cannot test in firefox on my vpn (getting that error)

you're probs getting the error cuz i changed my password & logged every device out of the acct as i thought we were done with the test lol

**Comment by mapx-** [comment]
No, I encountered the error in my first tests, so I gave up on Firefox and then tried in Chrome.

**Comment by AIDaN80112** [comment]
ah i see.

**Comment by AIDaN80112** [comment]
is there anyone with a mac that might be able to work on this?

**Comment by AIDaN80112** [comment]
update: it seems to have gone away. thanks for fixing!

**Comment by ComfortableMilk4454** [comment]
haven't used peacock in months but noticed today the detection issue is back: 

URL(s) where the issue occurs:

https://www.peacocktv.com/watch/playback/vod/GMO_00000000228248_02_UHDSDR/f927d9ea-4791-3410-b44f-8cbf6d4e2217

description: 

firefox release 128.0.3 (64-bit) on macOS 14.6 (23G80). uBO 1.59.0. happens on new firefox profile with only uBO installed with default settings.

other extensions used: 

![image](https://github.com/user-attachments/assets/26b9b03d-70c7-46cb-97b5-c164425ce9e7)

screenshots:

![image](https://github.com/user-attachments/assets/15dae476-5412-465b-aaf3-c57e17bf3be1)

Configuration:
<details>
<summary>Details</summary>


```yaml
uBlock Origin: 1.59.0
Firefox: 128
filterset (summary):
 network: 159241
 cosmetic: 118352
 scriptlet: 30656
 html: 2155
listset (total-discarded, last-updated):
 added:
  adguard-cookies: 30691-44, 1d.20h.33m
  ublock-cookies-adguard: 1284-14, 6d.20h.54m
  fanboy-cookiemonster: 67732-20241, 6d.20h.54m
  ublock-cookies-easylist: 1284-1284, 6d.20h.54m
  ublock-annoyances: 5929-66, 24m Δ
 default:
  user-filters: 0-0, never
  ublock-filters: 39661-123, 24m Δ
  ublock-badware: 10213-1, 24m Δ
  ublock-privacy: 1039-2, 24m Δ
  ublock-unbreak: 2529-0, 24m Δ
  ublock-quick-fixes: 136-0, 24m Δ
  easylist: 87793-173, 24m Δ
  easyprivacy: 53067-61, 24m Δ
  urlhaus-1: 28650-0, 13h.2m
  plowe-0: 3548-990, 6d.20h.54m
filterset (user): [empty]
userSettings: [none]
hiddenSettings: [none]
supportStats:
 allReadyAfter: 796 ms
 maxAssetCacheWait: 376 ms
 cacheBackend: indexedDB
popupPanel:
 blocked: 31
 network:
  conviva.com: 1
  cookielaw.org: 4
  demdex.net: 5
  freewheel.tv: 1
  mparticle.com: 2
  nbcuni.com: 1
  newrelic.com: 1
  nr-data.net: 2
  omtrdc.net: 12
  scorecardresearch.com: 2
 extended:
  ##.adsbox:remove()
```
</details>

**Comment by ComfortableMilk4454** [comment]
@mapx- if youre still part of the uBO team would you mind looking into [this](https://github.com/uBlockOrigin/uAssets/issues/21255#issuecomment-2269875665)? @ ing you cuz u fixed this issue the first time back in dec 2023, but its happening again. thanks so much!

**Comment by ComfortableMilk4454** [comment]
note that this also happens on firefox release 129.0 (since it came out this morning)

**Comment by ComfortableMilk4454** [comment]
this fixed itself!

**Comment by chkbm** [comment]
I am having this problem on Peacock where it's still showing ads, I've had this extension for over a year and it started showing ads 3 days ago. I am using Chrome on a Mac. I have looked extensively through all the other posts, 24773, 24243, etc. Which filter will prevent the ads? Something seems to have altered the extension.

**Comment by stephenhawk8054** [comment]
@BlakeButz We can't investigate without accounts. If you are willing to share account credentials, I'll give you an email.

**Comment by u-RraaLL** [comment]
There's a user at reddit saying they're willing to share: https://mod.reddit.com/mail/new/2q1r28

**Comment by kunalthapar** [comment]
@stephenhawk8054 I don't mind sharing my credentials.

**Comment by stephenhawk8054** [comment]
@kunalthapar Thanks. Can you send to this email: `stephenhawk8054@protonmail.com` ? Also, which country are you when accessing the site? I'll try with VPN too.

**Comment by stephenhawk8054** [comment]
@kunalthapar Can you test these filters:

```adb
peacocktv.com#@#.adsbox:remove()
||freewheel.tv^$image,redirect=1x1.gif,domain=peacocktv.com
||peacocktv.com/*/tracking/
peacocktv.com##+js(xml-prune, xpath(//*[name()="MPD"]/@mediaPresentationDuration | //*[name()="Period"]/@start | //*[name()="Period"][not(.//*[name()="SegmentTimeline"])][not(.//*[name()="ContentProtection"])] | //*[name()="Period"][./*[name()="BaseURL"]][not(.//*[name()="ContentProtection"])]), , .mpd)
```

Click on uBO icon > ⚙ [Dashboard button](https://github.com/gorhill/uBlock/wiki/Quick-guide:-popup-user-interface#the-tools) > Add the filter(s) in ["My filters" pane](https://github.com/gorhill/uBlock/wiki/Dashboard:-My-filters) > ✓ Apply changes > Open new tab and test again.

**Comment by kunalthapar** [comment]
@stephenhawk8054 That worked! Thank you!

**Comment by apetre070** [comment]
Hello team, 
Peacock seems to be freezing for me if I have uBO enabled on Firefox. I have tried updating the filters and also including the recommended filter above but it doesn't seem to have worked. 
Let me know how/if I can assist by providing more detail.

**Comment by stephenhawk8054** [comment]
https://github.com/uBlockOrigin/uAssets/issues/21255#issuecomment-3017896437

**Comment by apetre070** [comment]
@stephenhawk8054 This email address (from above)? `stephenhawk8054@protonmail.com` I can send creds

**Comment by stephenhawk8054** [comment]
Thanks. Can you send to that email?

**Comment by TheOneRichardAllan** [comment]
any new update lol? I'm still getting ads on Live feeds

**Comment by stephenhawk8054** [comment]
Of course there are no "updates" if there were no reports and no accounts.

https://github.com/uBlockOrigin/uAssets/issues/21255#issuecomment-3017896437

Don't just come here and ask for "updates" when we didn't receive any reports.

**Comment by u-RraaLL** [comment]
https://www.reddit.com/r/uBlockOrigin/comments/1pcrgef/ubo_preventing_peacocktvcom_stream_from_working/ + modmail

**Comment by u-RraaLL** [comment]
@stephenhawk8054  ?

https://mod.reddit.com/mail/all/327a86/4ehg4f

**Comment by stephenhawk8054** [comment]
I'll try to find some time. I'll be quite busy until the end of this month.

**Comment by CheeseburgerWithFries** [comment]
Ads issue is still present, no Detection issue noticed yet

I used the free Instacart+ trial --> free Peacock trial to make an account. No credit card is required for this, just an email. (Did not use a disposable email, but that might be possible.)

On loading an event, Peacock will play an unskippable 15 second ad, so far it's been the same movie trailer on every reload.

When the event takes a commercial break, ads will play. On one occasion I've seen a screen saying something like "Event in progress, will resume soon" with video of what I presume is [an indoor shopping gallery in Milan](https://en.wikipedia.org/wiki/Galleria_Vittorio_Emanuele_II) before ads resumed after about 10 seconds. I cannot tell whether the commercials are being injected directly into the video, or if they are an "overlay" on top of the "Event in progress" screen.

**Desired result:** No 15 second ad plays when loading a video. During regular commercials, only the "Event will resume soon" screen is shown.

Browser: Firefox 147.0.3 (64-bit) - Browser says it is up to date as of 2/7/2026

Extensions: Tested with all extensions disabled except for uBlock Origin. Issue is also present with the following extensions: LastPass, Feedbro, Enable Picture-in-Picture, Greasemonkey, Imagus, LibRedirect, RES

Filters: Tested with my custom filters both enabled and disabled

Attempted fix 1: Tried the custom filter from @stephenhawk8054 [shared earlier](https://github.com/uBlockOrigin/uAssets/issues/21255#issuecomment-3047052495) in this issue thread, but this did not work. Fix appears to be for an adblocker detection pop-up box though and not for live video ads.

Attempted fix 2: I've also tried starting the video with UBO disabled, then enabling once the video starts. This works on some other streaming sites, but did not work here.

<details>
<summary>Configuration/Troubleshooting Information</summary>

```yaml
uBlock Origin: 1.69.0
Firefox: 147
filterset (summary):
 network: 165611
 cosmetic: 42321
 scriptlet: 25954
 html: 3010
listset (total-discarded, last-updated):
 added:
  https://ublockorigin.github.io/uAssets/filters/resource-abuse.txt: 99-0, 2d.5h.46m
 default:
  user-filters: 49-6, never
  ublock-filters: 44270-72, 1h.55m Δ
  ublock-badware: 10836-31, 1h.55m Δ
  ublock-privacy: 3617-101, 1h.55m Δ
  ublock-unbreak: 2789-1, 1h.55m Δ
  ublock-quick-fixes: 440-13, 1h.55m Δ
  easylist: 90821-150, 1h.55m Δ
  easyprivacy: 55356-29, 1h.55m Δ
  urlhaus-1: 26660-5, 55m
  plowe-0: 3501-1003, 11d.3m
filterset (user): [array of 48 redacted]
trustedset:
 added: [array of 35 redacted]
switchRuleset:
 added: [array of 1 redacted]
userSettings: [none]
hiddenSettings: [none]
supportStats:
 allReadyAfter: 238 ms (selfie)
 maxAssetCacheWait: 60 ms
 cacheBackend: indexedDB
popupPanel:
 blocked: 906
 network:
  peacocktv.com: 334
  conviva.com: 89
  demdex.net: 1
  doubleclick.net: 1
  doubleverify.com: 1
  freewheel.tv: 1
  fwmrm.net: 8
  imrworldwide.com: 293
  innovid.com: 7
  nbcuas.com: 1
  newrelic.com: 1
  nr-data.net: 135
  omtrdc.net: 3
  scorecardresearch.com: 31
 extended:
  ##+js(xml-prune, xpath(//*[name()="MPD"]/@mediaPresentationDurat…

```

</details>    

Github says "You do not have permissions to reopen this issue," so I'm submitting as a comment and hoping someone with permissions sees it.
